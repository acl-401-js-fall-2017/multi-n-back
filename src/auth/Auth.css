.ReactModal__Content {
    position: relative;
}

.login-button {
    position: absolute;
    top: 53%; left: 50%;
    transform: translate(-50%, -50%);
    background-color: #254044;
    padding: 0.1em 1.2em;
    color: white;
    font-family: 'Pragati Narrow', sans-serif;
    font-size: 100%;
    outline: none;
    border: none;
    border-radius: 0.5em;
}

.login-button:hover {
    background-image: -webkit-gradient(radial, center, from(#d8eaec), to(end-color));
    background-image: -webkit-radial-gradient(center, #d8eaec, #ffffff00 25%, #ffffff00 75%, #d8eaec);
    background-image: -moz-radial-gradient(center, #d8eaec, #ffffff00 25%, #ffffff00 75%, #d8eaec);
    background-image: radial-gradient(at center, #ffffff00 60%, #d8eaec);
}

.close-modal-button {
    position: absolute;
    top: 15px; right: 15px;
    height: 2.3em;
    width: 2.3em;
    border-radius: 50%;
    border: none;
    line-height: 2em;
    text-align: center;
}

.login-window {
    z-index: 1000;
    position: fixed;
    top: 45%; left: 50%;
    transform: translate(-50%, -50%);
    min-width: 12em;
    padding: 2em 2em 1.4em 2em;
    background: white;
    border-radius: 1em;
}

.login-input-field {
    padding-bottom: 0.5em;
}

.login-input {
    position: relative;
    height: 1.2em;
}

.login-input label {
    position: absolute;
    left: 0;
}

.login-input input {
    position: absolute;
    right: 0;
    border: none;
    border-radius: 0.3em;
    background: #d8eaec;
    padding: 0.3em 0.3em 0.3em 1em;
    margin-top: -0.2em;
    letter-spacing: 0.1em;
    font-family: 'Pragati Narrow', sans-serif;
    font-size: 60%;    
    width: 7.5em;
    outline: none;
}

.login-input input:focus {
    border: 1px solid #568289;
}

.login-window p {
    margin: 0;
    font-size: 60%;
    text-align: right;
}

.login-buttons-field {
    display: flex;
    justify-content: space-around;
    margin-top: 0.5em;
}

.login-buttons-field button {
    border: none;
    border-radius: 0.8em;
    padding: 0.6em 1.4em 0.7em 1.4em;
    letter-spacing: 0.1em;
    font-family: 'Gruppo', cursive;
    font-weight: bold;
    font-size: 70%;
    color: white;
    box-sizing: border-box;
}

.signin-button {
    background-color: #568289;
}

.signup-button {
    background-color: #254044;
}

.login-buttons-field button:hover {
    box-shadow: 0 1px 3px #568289;
}








@Media (min-width: 1080px) {
    
    
}