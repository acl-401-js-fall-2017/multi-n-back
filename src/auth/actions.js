import * as auth from '../services/authApi';
import * as actions from '../app/constants';

export function signUp(name, password) {
  return async dispatch => {
    try {
      const token = await auth.post('/signup', { name, password });
      localStorage.setItem('token', token.token);
    }
    catch(error) {
      dispatch({
        type: actions.ERROR,
        payload: error
      });
    }
  };
}

export function signIn(name, password) {
  return async dispatch => {
    
    try {
      const token = await auth.post('/signin', { name, password });
      localStorage.setItem('token', token.token);
    }
    catch(error) {
      dispatch({
        type: actions.ERROR,
        payload: error
      });
    }
  };
}