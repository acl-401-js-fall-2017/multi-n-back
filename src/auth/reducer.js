import * as actions from '../app/constants';

export function user(store = null, { type, payload }) {
  switch (type) {
    case actions.SIGN_IN:
      return payload;
    case actions.LOG_OUT:
      return null;
    default:
      return store;
  }
}

export function userError(state = null, { type, payload }) {
  console.log('user error reducer');
  switch(type) {
    case actions.ERROR:
      return payload;
    case actions.LOADING:
      return null;
    default:
      return state;
  }
}