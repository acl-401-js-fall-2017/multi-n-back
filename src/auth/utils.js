//  use to reset a form within a component

// handleReset = () => {
//   const { user = {} } = this.props;
//   this.setState({
//     userName: user._id || null,
//     password: user.password
//   });
// };