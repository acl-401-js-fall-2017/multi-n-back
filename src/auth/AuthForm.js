import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { signUp, signIn } from './actions';
import { withRouter } from 'react-router-dom';
import Error from '../app/Error';

class AuthForm extends PureComponent {


  handleSignup = event => {
  
    if(event.target.value === 'signup') {
      this.props.signUp(event.target.parentNode.name.value, event.target.parentNode.password.value)
        .then(() => {
          if(!this.props.userError) {
            this.props.closeModal();
            this.props.history.push(this.props.location.pathname);
          }
        });
    }
  }

  handleSignin = event => {
    event.preventDefault();

    this.props.signIn(event.target.name.value, event.target.password.value)
      .then(() => {
        if(!this.props.userError) {
          this.props.closeModal();
          this.props.history.push(this.props.location.pathname);
        }
      });
  }

  handleLogout = () => {
    localStorage.removeItem('token');
    this.props.closeModal();
    this.props.history.push('/');
  }

  componentWillMount() {
    this.verifiedUser = !!localStorage.getItem('token');
  }

  render() {
    const { userError } = this.props;

    return (this.verifiedUser ? (
      <button onClick={this.handleLogout}>Logout</button>
    ) : (
      <form onClick={this.handleSignup} onSubmit={this.handleSignin}>
        <label>User Name:</label>
        <input name="name"/>
        <p>* user name must be unique</p>
        <label>Password:</label>
        <input type="password" name="password" placeholder="***************"/>
        <button type="submit" name="signin">SIGNIN</button>
        <button type="button" name="signup" value="signup" >SIGNUP</button>
        <Error error={userError}/>
      </form>
    ));
  }
}

export default withRouter(connect(
  state => ({ 
    userError: state.userError
  }),
  { signUp, signIn }
)(AuthForm));
