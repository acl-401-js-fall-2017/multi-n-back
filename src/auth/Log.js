import React, { PureComponent } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { logOut } from './actions';
import AuthForm from '../auth/AuthForm';


class Log extends PureComponent {
  constructor() {
    super();

    this.state = {
      modalIsOpen: false,
      confirmLogoutIsOpen: false
    };

    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  openModal() {
    this.setState({ modalIsOpen: true });
  }

  closeModal() {
    this.setState({ modalIsOpen: false });
  }

  handleLogoutToggle = () => this.setState({ confirmLogoutIsOpen: !this.state.confirmLogoutIsOpen })

  handleLogout = () => {
    localStorage.removeItem('token');
    this.handleLogoutToggle();
    this.props.logOut();
    this.props.history.push('/');
  }

  componentWillMount() {
    this.verifiedUser = !!localStorage.getItem('token');
  }

  render() {
    const { user } = this.props;

    return (
      <div>
        {user ? (
          <div className="user-options">
            <h5>hello, {user.name}</h5>
            <button className="logout-button" onClick={this.handleLogoutToggle}>Logout</button>
          </div>
        ) : (
          <div className="login-modal">
            <button className="login-button" onClick={this.openModal}>Login</button>
            {this.state.modalIsOpen &&
              <div className="login-window">
                <button className="close-modal-button" onClick={this.closeModal}>X</button>
                <AuthForm closeModal={this.closeModal} openModal={this.openModal}/>
              </div>
            }
          </div>
        )}
        {this.state.confirmLogoutIsOpen &&
          <section className="confirm-logout">
            <button className="confirm-logout-close" onClick={this.handleLogoutToggle}>X</button>
            <p>are you sure?</p>
            <button className="confirm-logout-button" onClick={this.handleLogout}>yes</button>
          </section>
        }
      </div>
    );
  }
}
export default withRouter(connect(
  state => ({ user: state.user }),
  { logOut }
)(Log));
