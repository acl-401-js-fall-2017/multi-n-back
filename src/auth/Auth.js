import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import Log from './Log';

class Auth extends PureComponent {
 
  render() {
    return (
      <div className="auth">
        <h3>Auth component</h3>
        <Log/>
      </div>
    );
  }
}

export default connect(
  state => ({}),
  null
)(Auth);