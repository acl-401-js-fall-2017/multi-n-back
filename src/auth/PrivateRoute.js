import React from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import Log from './Log';

const PrivateRoute = ({ component: Component, render, ...rest }) => (
  <Route {...rest} render={props => {

    if(localStorage.getItem('token')) {
      return render ? render(props) : <Component {...props}/>;
    }
    else {
      return (
        <Log/>
      );
    }
  }}/>
);

export default connect(
  state => ({}),
  null
)(PrivateRoute);