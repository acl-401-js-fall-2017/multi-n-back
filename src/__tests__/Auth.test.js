import React from 'react';
import { shallow } from 'enzyme';
import { Auth } from '../auth/Auth';

describe('Auth', () => {
  it('shows Auth component', () => {

    const wrapper = shallow(
      <Auth />
    );
    expect(wrapper).toMatchSnapshot();
  });
});