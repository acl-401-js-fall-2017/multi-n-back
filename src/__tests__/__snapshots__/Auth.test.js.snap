// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth shows Auth component 1`] = `
<div
  className="auth"
>
  <withRouter(Connect(Log)) />
</div>
`;
